core/test/base/array.cpp

```diff
@@ -284,6 +284,17 @@ TYPED_TEST(Array, CanBeResized)
 }
 
 
+TYPED_TEST(Array, ViewCanBeResetNotResized)
+{
+    TypeParam data[] = {1, 2, 3};
+    auto view = gko::Array<TypeParam>::view(this->exec, 3, data);
+    view.resize_and_reset(1);
```

Comment:
Yes it becomes an array, because we do the following:
```cpp
Array tmp(std::move(exec));
tmp = *this;
```
The first line creates an **array** (with `default_deleter`) in the other executor. Then we copy the data to it which ensures that both objects keep their nature (`tmp` will stay an array after this copy). Thanks for this question actually I hadn't checked to make sure everything is proper!

About the `resize_and_reset`, I am still unsure what we should do with it. Maybe we should discuss it in today's call. Tobias at some point wanted it to change the nature of the object (view would become an array transparently), you want to disable it which makes sense, I am also fine with the current `clear()` version (the question is whether resizing an array or not is an error here, it can be considered both an error or not one depending on POV). I don't have a strong opinion on this, but picking something is tough because there are many valid options.